src/main/scala/Actor.scala

```diff
@@ -119,19 +119,19 @@ case class Actor(
     rookPath = rookPos <-> newRookPos
     mustBeUnoccupied = (kingPath ++ rookPath).filter(_ != kingPos).filter(_ != rookPos)
     if !mustBeUnoccupied.exists(board.pieces.contains)
-    if !kingPath.exists(p => board.variant.kingThreatened(board, !color, p))
-    b1 <- board take rookPos
-    b2 <- newKingPos match {
-      case p if p == kingPos => Some(b1)
-      case p => b1.move(kingPos, p)
-    }
-    b3 <- b2.place(color.rook, newRookPos)
-    b4 = b3 updateHistory (_ withoutCastles color)
+    b1 <- board take kingPos
+    mustNotBeAttacked = kingPath.filter(_ != newKingPos || kingPos == newKingPos)
+    if !mustNotBeAttacked.exists(p => board.variant.kingThreatened(b1, !color, p))
+    b2 <- b1 take rookPos
+    b3 <- b2.place(color.king, newKingPos)
+    b4 <- b3.place(color.rook, newRookPos)
+    if !board.variant.kingThreatened(b4, !color, newKingPos)
```

Comment:
Finally test the destination king square seperately, as if the rook has already moved.